### PR TITLE
fix: File uploads should only be allowed for room members

### PR DESCRIPTION
**Pull request by MartinSchoeler**
## Proposed changes (including videos or screenshots)

Currently non channel members can drag and drop files into a channel they do not belong, this PR fixes that

## Issue(s)

[CORE-205](https://rocketchat.atlassian.net/browse/CORE-205)

[CORE-205]: https://rocketchat.atlassian.net/browse/CORE-205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/views/room/body/hooks/useFileUploadDropTarget.ts [hunk 1]
```diff
@@ -78,12 +79,20 @@ export const useFileUploadDropTarget = (): readonly [
 			} as const;
 		}
 
+		if (!subscription) {
+			return {
+				enabled: false,
+				reason: t('error-not-allowed'),
+				...overlayProps,
+			} as const;
+		}
```
Comment (dougfabris) [inline apps/meteor/client/views/room/body/hooks/useFileUploadDropTarget.ts:88]:
```suggestion
		if (!fileUploadAllowedForUser || !subscription) {
			return {
				enabled: false,
				reason: t('error-not-allowed'),
				...overlayProps,
			} as const;
		}
```

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: apps/meteor/tests/e2e/file-upload.spec.ts [hunk 2]
```diff
@@ -76,3 +76,27 @@ test.describe.serial('file-upload', () => {
 		await expect(poHomeChannel.content.btnModalConfirm).not.toBeVisible();
 	});
 });
+test.describe('file-upload-not-member', () => {
```
Comment (KevLehman) [inline apps/meteor/tests/e2e/file-upload.spec.ts:79]: Should we check how these behaves on Omni rooms?

Comment (MartinSchoeler) [reply apps/meteor/tests/e2e/file-upload.spec.ts:79]: I don`t think so, they behave exactly the same as the other channels in this regard, the agent/manager cannot send messages in the room unless they are a part of it

Comment (dougfabris) [reply apps/meteor/tests/e2e/file-upload.spec.ts:79]: @MartinSchoeler indeed, not necessary!

**Review by dougfabris — APPROVED** [review]
This is such a complete work, congrats on that!

File: apps/meteor/tests/e2e/file-upload.spec.ts (hunk 2 above)
Comment (KevLehman) [reply apps/meteor/tests/e2e/file-upload.spec.ts:79]: noice noice

**Review by KevLehman — APPROVED** [review]

**Pull request merged**